cmd/controller-gen/main.go

```diff
@@ -84,7 +84,7 @@ Usage:
 }
 
 func newCRDCmd() *cobra.Command {
-	g := &crdgenerator.Generator{}
+	o := &crd.Options{}
```

Comment:
If the pkg name becomes `crd` then `Generator` seems like more appropriate to capture generator options ?

Reply:
Make sense. So we can keep generator approach, but just split the generation procedure into some exposed APIs. I also wanted to have a more generic `Generator` upfront, which wraps `gengo` for all kinds of generation in this repo. This kind of `Generator` can live in `pkg/internal/codegen`